### PR TITLE
docs - awscc_ec2_transit_gateway_route_table_association

### PR DESCRIPTION
**Pull request by hlazrv**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

File: docs/resources/ec2_transit_gateway_route_table_association.md [hunk 1]
```diff
@@ -10,6 +9,60 @@ description: |-
...
+
+#Create a transit gateway attachment
+resource "awscc_ec2_transit_gateway_attachment" "transit_gateway_attachment_example" {
+  subnet_ids         = [aws_subnet.example.id]
+  transit_gateway_id = awscc_ec2_transit_gateway.transit_gateway_example.id
+  vpc_id             = aws_vpc.example.id
+}
+
+# Create an VPC
+resource "aws_vpc" "example" {
```
Comment (quixoticmonk) [inline docs/resources/ec2_transit_gateway_route_table_association.md:31]: Thank you , @hlazrv . 
- Can we use the `awscc_ec2_vpc` and `awscc_ec2_subnet` in place of the AWS provider resource here ?
- (nit): You can shorten the resource names to `example` unless there are multiple resources in the same configuration.

File: examples/resources/awscc_ec2_transit_gateway_route_table_association/ec2_transit_gateway_route_table_association.tf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# Create a public subnet 
+resource "aws_subnet" "example" {
+  vpc_id     = aws_vpc.example.id
+  cidr_block = "10.0.0.0/24"
+  tags = {
+    Name = "PublicSubnet"
+  }
+
+  depends_on = [aws_vpc.example]
```
Comment (quixoticmonk) [inline examples/resources/awscc_ec2_transit_gateway_route_table_association/ec2_transit_gateway_route_table_association.tf:32]: Is the depends_on needed with the vpc_id assignment ?

File: templates/resources/ec2_transit_gateway_route_table_association.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+---
+
+# {{.Name}} ({{.Type}})
+
+{{ .Description | trimspace }}
+
+## Example Usage
+
+### First example
```
Comment (quixoticmonk) [inline templates/resources/ec2_transit_gateway_route_table_association.md.tmpl:14]: (nit): We can remove the `### First example` unless we intend to add any more to the PR. The statement below would help deliver the usecase.

File: docs/resources/ec2_transit_gateway_route_table_association.md (hunk 1 above)
Comment (hlazrv) [reply docs/resources/ec2_transit_gateway_route_table_association.md]: Updated

File: templates/resources/ec2_transit_gateway_route_table_association.md.tmpl (hunk 3 above)
Comment (hlazrv) [reply templates/resources/ec2_transit_gateway_route_table_association.md.tmpl]: Removed

File: examples/resources/awscc_ec2_transit_gateway_route_table_association/ec2_transit_gateway_route_table_association.tf (hunk 2 above)
Comment (hlazrv) [reply examples/resources/awscc_ec2_transit_gateway_route_table_association/ec2_transit_gateway_route_table_association.tf]: Removed

**Review by quixoticmonk — APPROVED** [review]
LGTM

**Comment by marcosentino** [comment]
@hlazrv Thank you for the contribution!

**Pull request merged**